MapboxCoreNavigation/OfflineDirections.swift

```diff
@@ -0,0 +1,194 @@
+    public typealias UnpackCompletionHandler = (_ result: UInt64, _ error: Error?) -> ()
+    
+    public override init(accessToken: String? = nil, host: String? = nil) {
+        super.init(accessToken: accessToken, host: host)
+    }
+    
+    public func configureRouter(tilesURL: URL, translationsURL: URL? = nil, completionHandler: @escaping OfflineDirectionsCompletionHandler) {
+        
+        OfflineDirectionsConstants.offlineSerialQueue.sync {
+            // Translations files bundled winthin navigation native
+            // will be used when passing an empty string to `translationsPath`.
+            let tileCount = self.navigator.configureRouter(forTilesPath: tilesURL.path, translationsPath: translationsURL?.path ?? "")
+            
+            DispatchQueue.main.async {
+                completionHandler(tileCount)
+            }
+        }
+    }
+    
+    public class func unpackTilePack(at filePathURL: URL, outputDirectoryURL: URL, progressHandler: UnpackProgressHandler?, completionHandler: UnpackCompletionHandler?) {
+        
+        OfflineDirectionsConstants.offlineSerialQueue.sync {
+            
+            let totalPackedBytes = filePathURL.fileSize!
+            
+            // Report 0% progress
+            progressHandler?(totalPackedBytes, totalPackedBytes)
+            
+            var timer: DispatchTimer? = DispatchTimer(countdown: .seconds(500), accuracy: .seconds(500), executingOn: OfflineDirectionsConstants.unpackSerialQueue) {
+                if let remainingBytes = filePathURL.fileSize {
+                    progressHandler?(totalPackedBytes, remainingBytes)
+                }
+            }
+
+            timer?.arm()
+            
+            let tilePath = filePathURL.path
+            let outputPath = outputDirectoryURL.path
+            
+            let result = MBNavigator().unpackTiles(forPacked_tiles_path: tilePath, output_directory: outputPath)
+            
+            // Report 100% progress
+            progressHandler?(totalPackedBytes, totalPackedBytes)
+            
+            timer?.disarm()
+            timer = nil
+            
+            DispatchQueue.main.async {
+                completionHandler?(result, nil)
+            }
+        }
+    }
+    
+    public func calculate(_ options: RouteOptions, offline: Bool = true, completionHandler: @escaping Directions.RouteCompletionHandler) {
+        
+        guard offline == true else {
+            super.calculate(options, completionHandler: completionHandler)
+            return
+        }
+        
+        let url = self.url(forCalculating: options)
+        
+        OfflineDirectionsConstants.offlineSerialQueue.async { [weak self] in
+            
+            guard let result = self?.navigator.getRouteForDirectionsUri(url.absoluteString) else {
+                let error = OfflineRoutingError.unexpectedRouteResult("Unexpected routing result")
```

Comment:
We should make these error messages localizable, since they end up in `Error.localizedDescription` and would likely be presented to the user in an alert dialog.